<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Summary • PhyInsight</title><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Summary"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">PhyInsight</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/PhyloTree_vignette_1.html">Quick Tree Building</a>
    <a class="dropdown-item" href="articles/PhyloTree_vignette_2.html">Automated String Clean Up</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-title-body">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Summary</h1>
      
    </div>


<p> </p>
<div id="summary" class="section level1">

<p><code>PhyInsight</code> is an open-source R package [@R] designed to automate the complicated process of DNA string manipulation and phylogenetic tree creation. This tool provides functions that enable the user to query specimen data, manipulate DNA strings, and create phylogenetic trees effortlessly. This tool was created for beginners who are new to bioinformatics and phylogeny, as well as for experts who will benefit from the streamlined workflow, especially during iterative tasks in exploratory and confirmatory data analysis.</p>
<p> </p>
</div>
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a></h1>
<p>While existing bioinformatic tools in phylogeny are powerful, they require much work to prepare the data for analysis and phylogenetic tree creation. This hinders reproducibility and uses up valuable research time. By reducing complex processes into a few easy-to-use functions, this package provides automated solutions for DNA string manipulation and phylogenetic tree creation within the R environment [@R]. In doing so, this package reduces the risk of errors, promotes reliability, and saves the user time. Furthermore, this package allows the user to perform complicated analyses without the need for extensive phylogenetic or programming knowledge making it perfect for beginners as well as experts.</p>
<p> </p>
<p><code>PhyInsight</code> incorporates various open-source packages, streamlining and facilitating a wide range of processes essential for phylogenetic analysis. The library includes features from <code>bold</code> [@bold] for querying specimen data, <code>ape</code> [@ape] for DNA distance calculation, and both <code>ape</code> and <code>Biostrings</code> [@Biostrings] for object coercion. Additionally, it incorporates features from <code>DECIPHER</code> [@DECIPHER] for the orientation and alignment of DNA strings. In short, <code>PhyInsight</code> serves as an efficient tool that brings together functionalities, making phylogenetic analysis more efficient and user-friendly.</p>
<p> </p>
<p>Version 0.1.0 is available at the Comprehensive R Archive Network (CRAN) and is maintained on <a href="https://github.com/JamesC845/PhyInsight" class="external-link">GitHub</a>. Furthermore, a <a href="https://jamesc845.github.io/PhyInsight/">pkgdown website</a> is available to view its functions and vignettes.</p>
<p> </p>
</div>
<div class="section level1">
<h1 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a></h1>
<p>The <code>PhyInsight</code> package uses three functions to prepare a DNA string set once the specimen data is ready. For conciseness, the code querying and subsetting the specimen data was left out of the example. However, an example of this task can be seen in a <a href="https://jamesc845.github.io/PhyInsight/articles/PhyloTree_vignette_2.html">vignette</a> on the website. There are three variations of a comprehensive function for string clean-up that can be used to automatically remove problem strings from a string set should the need arise.</p>
<p> </p>
<div class="section level4">
<h4 id="create-dna-string-set">Create DNA String Set<a class="anchor" aria-label="anchor" href="#create-dna-string-set"></a></h4>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># generate a DNA bin using the specimen data frame</span></span>
<span><span class="va">DNABin</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/genDNABin.html">genDNABin</a></span><span class="op">(</span><span class="va">specimen_dataframe</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># use the DNA bin to create a DNA string set</span></span>
<span><span class="va">DNAStringSet</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/genDNAStringSet.html">genDNAStringSet</a></span><span class="op">(</span><span class="va">DNABin</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># automatically manipulate the DNA strings </span></span>
<span><span class="va">DNAStringSet_manip</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/ManipStringSet.html">ManipStringSet</a></span><span class="op">(</span><span class="va">DNAStringSet</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="string-clean-up">String Clean Up<a class="anchor" aria-label="anchor" href="#string-clean-up"></a></h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># use the function to automatically remove problem strings from the DNA string set</span></span>
<span><span class="va">BadStringsRemoved</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/rmBadStrings_3.html">rmBadStrings_3</a></span><span class="op">(</span></span>
<span>  <span class="va">DNAStringSet_manip</span>,</span>
<span>  <span class="va">specimen_dataframe</span>,</span>
<span>  rmOutliers <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  max_Z_score <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># store the cleaned data into new objects</span></span>
<span><span class="va">DNAStringSet_clean</span> <span class="op">&lt;-</span> <span class="va">BadStringsRemoved</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="va">specimen_dataframe_clean</span> <span class="op">&lt;-</span> <span class="va">BadStringsRemoved</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></span>
<span></span>
<span><span class="co"># automatically create a phylo tree</span></span>
<span><span class="va">PhyloTree</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/genPhytree.html">genPhytree</a></span><span class="op">(</span><span class="va">DNAStringSet_clean</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="save-phylogenetic-tree">Save Phylogenetic Tree<a class="anchor" aria-label="anchor" href="#save-phylogenetic-tree"></a></h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/savePhytree.html">savePhytree</a></span><span class="op">(</span></span>
<span>  <span class="va">PhyloTree</span>,</span>
<span>  <span class="va">specimen_dataframe_clean</span>,</span>
<span>  tree_file_name <span class="op">=</span> <span class="st">"figure.pdf"</span>,</span>
<span>  openPDF <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  savePDF <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  label_offset <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="figure.jpg" alt=""><p class="caption"></p>
</div>
<p> </p>
</div>
</div>
<div class="section level1">
<h1 id="improvement">Improvement<a class="anchor" aria-label="anchor" href="#improvement"></a></h1>
<p>In the future, the package will continue to improve and be updated. Currently, intended updates include:</p>
<ul><li><p>methods for automated phylogenetic inference will be added</p></li>
<li><p>queried data will be compiled from multiple databases rather than solely from Bold Systems</p></li>
<li><p>functions for querying, writing, and reading FASTA files will be provided</p></li>
</ul><p> </p>
</div>
<div class="section level1">
<h1 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h1>
</div>


  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by James Churchward.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

